src/shared/utils/helpers/get-community-details.ts

```diff
@@ -0,0 +1,21 @@
+import { Community } from "lemmy-js-client";
+import hostname from "./hostname";
+
+export default function getCommunityDetails(
+  community: Community
+): [string, string, string] {
+  let name_: string, title: string, link: string;
+  const local = community.local == null ? true : community.local;
+  if (local) {
+    name_ = community.name;
+    title = community.title;
+    link = `/c/${community.name}`;
+  } else {
+    const domain = hostname(community.actor_id);
+    name_ = `${community.name}@${domain}`;
+    title = `${community.title}@${domain}`;
+    link = !this.props.realLink ? `/c/${name_}` : community.actor_id;
+  }
+
+  return [name_, title, link];
```

Comment:
Nitpick. Could be worth annotating the function return type as `[string, string, string]` since the return is meant to be a tuple instead of a list of strings that could vary in length.

Reply:
do you mean a change in the function return type `getCommunityDetails(...): [string, string, string] {`

or this return value being cast here, something like

`return [name_, title, link] as [string, string, string];` or

`return [name_, title, link] as const` (I think both work)

Reply:
> do you mean a change in the function return type getCommunityDetails(...): [string, string, string] {

That one.